src/main/java/org/apache/sysds/hops/rewrite/RewriteConstantFolding.java

```diff
@@ -98,13 +98,7 @@ private Hop rConstantFoldingExpression( Hop root ) {
 		if( root.getDataType() == DataType.SCALAR //scalar output
 			&& ( isApplicableBinaryOp(root) || isApplicableUnaryOp(root) ) )
 		{ 
-			//core constant folding via runtime instructions
-			try {
-				literal = evalScalarOperation(root); 
-			}
-			catch(Exception ex) {
-				LOG.error("Failed to execute constant folding instructions. No abort.", ex);
-			}
+			literal = evalScalarOperation(root); 
```

Comment:
It would anyway fail at some point not only if it tries to rewrite, right? Did you check if the error message still is informative enough if it fails here?

Reply:
The main issue is that previously it did not fail. because the error was caught and not handled, while the program would crash at some arbitrary place afterwards (or not at all). When the system crashed the exception had no reference to this place unless your logging settings were set to ERROR level and in that case you had to still sift through the logging messages to find the exception. that was the actual root course. (to much work for a single exception)

Reply:
If it did not fail was the result as expected? Anyway if the error is more obvious this way then this seems better, although it is strange that the rewrite fails even though the code runs.